crates/node/src/interactive_cli/identity.rs

```diff
@@ -75,59 +77,58 @@ enum AliasSubcommands {
 }
 
 impl IdentityCommand {
-    pub fn run(self, node: &Node) -> EyreResult<()> {
+    pub async fn run(self, node_client: &NodeClient, ctx_client: &ContextClient) -> EyreResult<()> {
         let ind = ">>".blue();
 
         match self.subcommand {
             IdentitySubcommands::List { context } => {
-                list_identities(node, context, &ind.to_string())?;
+                list_identities(node_client, ctx_client, context, &ind.to_string()).await?;
             }
             IdentitySubcommands::New => {
-                create_new_identity(node, &ind.to_string());
+                create_new_identity(ctx_client, &ind.to_string());
             }
             IdentitySubcommands::Alias { command } => {
-                handle_alias_command(node, command, &ind.to_string())?;
+                handle_alias_command(node_client, command, &ind.to_string())?;
             }
         }
 
         Ok(())
     }
 }
 
-fn list_identities(node: &Node, context: Alias<ContextId>, ind: &str) -> EyreResult<()> {
-    let context_id = node
-        .ctx_manager
+async fn list_identities(
+    node_client: &NodeClient,
+    ctx_client: &ContextClient,
+    context: Alias<ContextId>,
+    ind: &str,
+) -> EyreResult<()> {
+    let context_id = node_client
         .resolve_alias(context, None)?
         .ok_or_eyre("unable to resolve")?;
 
-    let handle = node.store.handle();
-    let mut iter = handle.iter::<ContextIdentityKey>()?;
+    println!("{ind} {:44} | {}", "Identity", "Owned");
+
+    // Get the stream of identities
+    let stream = ctx_client.context_members(&context_id, None).await;
+    let stream_owned = ctx_client.context_members(&context_id, Some(true)).await;
 
-    let first = 'first: {
-        let Some(k) = iter
-            .seek(ContextIdentityKey::new(context_id, [0; 32].into()))
-            .transpose()
-        else {
-            break 'first None;
-        };
+    // Pin the streams
+    futures_util::pin_mut!(stream);
+    futures_util::pin_mut!(stream_owned);
```

Comment:
whoops, that's an unintended effect of the API change..

how about `context_members` returns `(public_key, is_owned)`?

since we already know that information within the function